src/dune/install_rules.ml

```diff
@@ -390,34 +352,33 @@ let init_meta sctx ~dir =
             })
         ~else_:(Build.return ["# DUNE_GEN"])
     in
-    let meta_contents =
-      version >>^ fun version ->
-      Gen_meta.gen
-        ~package:(Package.Name.to_string pkg.name)
-        ~version
-        (Lib.Local.Set.to_list libs
-         |> List.map ~f:Lib.Local.to_lib)
-    in
     let ctx = Super_context.context sctx in
     Super_context.add_rule sctx ~dir:ctx.build_dir
-      (Build.fanout meta_contents template
-       >>^ (fun ((meta : Meta.t), template) ->
-         let buf = Buffer.create 1024 in
-         let ppf = Format.formatter_of_buffer buf in
-         Format.pp_open_vbox ppf 0;
-         List.iter template ~f:(fun s ->
-           if String.is_prefix s ~prefix:"#" then
-             match
-               String.extract_blank_separated_words (String.drop s 1)
-             with
-             | ["JBUILDER_GEN" | "DUNE_GEN"] ->
-               Format.fprintf ppf "%a@," Meta.pp meta.entries
-             | _ -> Format.fprintf ppf "%s@," s
-           else
-             Format.fprintf ppf "%s@," s);
-         Format.pp_close_box ppf ();
-         Format.pp_print_flush ppf ();
-         Buffer.contents buf)
+      (let open Build.S.O in
+       (let+ template = template in
+        let meta =
+          Gen_meta.gen
+            ~package:(Package.Name.to_string pkg.name)
+            ~version:pkg.version
+            (Lib.Local.Set.to_list libs
+             |> List.map ~f:Lib.Local.to_lib)
+        in
+        let buf = Buffer.create 1024 in
+        let ppf = Format.formatter_of_buffer buf in
+        Format.pp_open_vbox ppf 0;
+        List.iter template ~f:(fun s ->
+          if String.is_prefix s ~prefix:"#" then
+            match
+              String.extract_blank_separated_words (String.drop s 1)
+            with
+            | ["JBUILDER_GEN" | "DUNE_GEN"] ->
```

Comment:
Shall we drop support for `JBUILDER_GEN`?

Reply:
Well, we should at least raise, otherwise it'd be easy for such an error to go unnoticed. It feels simpler to keep the code as-is. The `JBUILDER_GEN` support is not very expansive here